source/simulator/assembly.cc

```diff
@@ -1419,7 +1419,8 @@ namespace aspect
                    * on the direction of the flow. Thus the flow-dependent terms below are
                    * only calculated if the edge is an inflow edge.
                    */
-                  const bool inflow = ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.);
+                  const bool inflow = (advection_field.is_temperature()
+                                       || ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.));
```

Comment:
Since we are in the convection-dominated regime in general, we should philosophically be imposing the Dirichlet values via the flux terms. The jump terms come more from the pure diffusion elliptic limit, and so are relatively unimportant here. Neglecting the flow jumps would probably be a bad idea.